src/user-post/service/post.service.ts

```diff
@@ -0,0 +1,188 @@
+import { Injectable } from '@nestjs/common';
+import { Posts } from '../entities/Post.entity';
+import { EditPostDto, PostDto } from '../dto/post.dto/post.dto';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { User } from '../../authentication/entities/user.entity';
+import { FirebaseApp } from '../../Firebase/firebase.service';
+import { NotFoundError } from 'rxjs';
+
+@Injectable()
+export class PostService {
+  constructor(
+    @InjectRepository(Posts) private postsRepository: Repository<Posts>,
+    @InjectRepository(User) private userRepository: Repository<User>,
+    private readonly firebaseService: FirebaseApp,
+  ) {}
+  async createPost(
+    post: PostDto,
+    photo: Express.Multer.File,
+  ): Promise<Posts[]> {
+    const userId = await this.userRepository.findOne({
+      where: {
+        userIdToken: post.userId,
+      },
+    });
+    const newPost: Posts = this.postsRepository.create({
+      title: post.title,
+      description: post.description,
+      photo: photo ? await this.firebaseService.uploadFile(photo) : null,
+      userId: userId.id,
+    });
+    await this.postsRepository.save(newPost);
+    const createdPost: Posts[] = await this.postsRepository.find({
+      where: {
+        id: newPost.id,
+      },
+      relations: {
+        user: true,
+      },
+      order: {
+        createdAt: 'DESC',
+      },
+    });
+    return createdPost.reverse();
+  }
+  async getPosts(userId: string): Promise<Posts[]> {
+    const getUser: User = await this.userRepository
+      .findOne({
+        where: {
+          userIdToken: userId,
+        },
+      })
+      .catch(() => {
+        throw new Error('Something goes wrong');
+      });
```

Comment:
Maybe we should use traditional try-catch block? Please use more meaningful error message

Reply:
done